modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceRegionLocalServiceImpl.java

```diff
@@ -178,27 +189,118 @@ public CommerceRegion setActive(long commerceRegionId, boolean active)
 
 	@Override
 	public CommerceRegion updateCommerceRegion(
-			long commerceRegionId, String name, String code, double priority,
-			boolean active, ServiceContext serviceContext)
+			long commerceRegionId, Map<Locale, String> nameMap, String code,
+			double priority, boolean active, ServiceContext serviceContext)
 		throws PortalException {
 
 		CommerceRegion commerceRegion =
 			commerceRegionPersistence.findByPrimaryKey(commerceRegionId);
 
-		validate(name);
+		validate(nameMap);
 
-		commerceRegion.setName(name);
 		commerceRegion.setCode(code);
 		commerceRegion.setPriority(priority);
 		commerceRegion.setActive(active);
 
+		// Update CommerceRegion localization
+
+		_updateCommerceRegionLocalizedFields(
+			commerceRegion.getCompanyId(), commerceRegionId, nameMap);
+
 		return commerceRegionPersistence.update(commerceRegion);
 	}
 
-	protected void validate(String name) throws PortalException {
-		if (Validator.isNull(name)) {
+	protected void validate(Map<Locale, String> nameMap)
+		throws PortalException {
+
+		if (nameMap == null) {
 			throw new CommerceRegionNameException();
 		}
 	}
 
+	private List<CommerceRegionLocalization> _addCommerceRegionLocalizedFields(
+		long companyId, long commerceRegionId, Map<Locale, String> nameMap) {
+
+		Set<Locale> localeSet = new HashSet<>();
+
+		if (nameMap != null) {
+			localeSet.addAll(nameMap.keySet());
+		}
+
+		List<CommerceRegionLocalization> commerceRegionLocalizations =
+			new ArrayList<>();
```

Comment:
Minor sf, but we can move `List<CommerceRegionLocalization> commerceRegionLocalizations = new ArrayList<>();` before `Set<Locale> localeSet = new HashSet<>();` since localSet is only being used to modify commerceRegionLocalizations.